lib/Dialect/TritonGPU/IR/Dialect.cpp

```diff
@@ -991,6 +991,7 @@ unsigned SharedEncodingAttr::getTotalElemsPerThread(ArrayRef<int64_t> shape,
   llvm_unreachable("getElemsPerThread is not supported for shared layout");
   return 0;
 }
+int32_t SharedEncodingAttr::getAlignment() const { return 128 * getMaxPhase(); }
```

Comment:
This is correct for the hasLeadingOffset encodings, not entirely sure about the others.

Reply:
Seems too large for encodings without a leading offset

Reply:
I don’t know where 128 comes from specifically when hasLeadingOffset=False

Reply:
I think for cases with `hasLeadingOffset=False` we only need an alignment of 16.

this is a nit but I wonder if it makes sense to have alignment be a property of the encoding? I this logic could be moved in a builder for AllocOp